File: backend/app/schemas/alert.py
from datetime import datetime

from pydantic import BaseModel

from app.models.alert_rule import AlertType


class AlertRuleOut(BaseModel):
    id: int
    alert_type: AlertType
    threshold_value: float | None
    cooldown_minutes: int
    is_enabled: bool
    notify_telegram: bool
    notify_email: bool
    description: str | None

    class Config:
        from_attributes = True


class AlertRuleUpdate(BaseModel):
    threshold_value: float | None = None
    cooldown_minutes: int | None = None
    is_enabled: bool | None = None
    notify_telegram: bool | None = None
    notify_email: bool | None = None


class NotificationLogOut(BaseModel):
    id: int
    router_id: int | None
    alert_type: str
    target_identifier: str | None
    message: str
    channel: str
    resolved: bool
    sent_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/alert.ts
export type AlertType = "router_down" | "vpn_down" | "cpu_high" | "mem_high" | "isp_down";

export interface AlertRule {
  id: number;
  alert_type: AlertType;
  threshold_value: number | null;
  cooldown_minutes: number;
  is_enabled: boolean;
  notify_telegram: boolean;
  notify_email: boolean;
  description: string | null;
}

export interface AlertRuleUpdatePayload {
  threshold_value?: number;
  cooldown_minutes?: number;
  is_enabled?: boolean;
  notify_telegram?: boolean;
  notify_email?: boolean;
}

export interface NotificationLogEntry {
  id: number;
  router_id: number | null;
  alert_type: string;
  target_identifier: string | null;
  message: string;
  channel: string;
  resolved: boolean;
  sent_at: string;
}
